courses/C/intermediate-level/Dynamic-memory-allocation/Intro.md

```diff
@@ -0,0 +1,112 @@
+---
+id: lesson-1
+title: "Dynamic Memory Allocation in C"
+sidebar_label: Dynamic Memory Allocation
+sidebar_position: 1
+description: "Dynamic Memory Allocation in C"
+tags: [courses,intermediate-level,Dynamic Memory Allocation,Introduction]
+---
+ 
+
+Dynamic memory allocation allows programs to allocate memory at runtime, which is useful for creating flexible and efficient programs. The C Standard Library provides several functions for dynamic memory management.
+
+#### Introduction to Dynamic Memory Allocation
+
+Dynamic memory allocation is the process of allocating memory storage during the execution of a program. This is done using pointers to manage memory blocks of varying sizes.
+
+#### Flowchart: Dynamic Memory Allocation
+:::note
+1. **Start**
+2. **Allocate memory** using `malloc` or `calloc`
+3. **Check if allocation was successful**
+   - If successful, proceed to the next step
+   - If not, print an error message and **exit**
+4. **Use the allocated memory** (assign values, process data)
+5. **Free the allocated memory** using `free`
+6. **End**
+:::
+
+```
+ +--------------------------+
+ | Start                    |
+ +--------------------------+
+            |
+            v
+ +--------------------------+
+ | Allocate memory using    |
+ | malloc or calloc         |
+ +--------------------------+
+            |
+            v
+ +--------------------------+
+ | Check if allocation was  |
+ | successful               |
+ +--------------------------+
+            |
+          /   \
+       Yes     No
+        |       |
+        v       v
++------------------+  +--------------------------+
+| Use the allocated|  | Print error message and |
+| memory           |  | exit                    |
++------------------+  +--------------------------+
+        |
+        v
++--------------------------+
+| Free the allocated memory|
+| using free               |
++--------------------------+
+        |
+        v
++--------------------------+
+| End                      |
++--------------------------+
+```
+
```

Comment:
You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)